mock: bind mount qubes-pesign socket on build

fepitre opened the pull request:
https://github.com/QubesOS/qubes-issues/issues/8206

marmarek commented on qubesbuilder/plugins/chroot_rpm/mock/fedora.cfg:
```diff
@@ -17,6 +17,10 @@ config_opts['use_bootstrap'] = False
 
+# mount the pesign socket into the chroot
+config_opts['plugin_conf']['bind_mount_opts']['dirs'].append(('/var/run/qubes-pesign', '/var/run/qubes-pesign' ))
+config_opts['nspawn_args'] += ['--bind=/var/run/qubes-pesign']
```
Aren't those two meaning the same thing? Why both are needed?

fepitre replied:
I've copied paste some snippet in a thread but I've tested and only the second one is needed.

fepitre committed:
mock: bind mount qubes-pesign socket on build

marmarek commented:
@fepitre what is the recommended method for redirecting `/var/run/qubes-pesign` to qrexec? socat started in `/rw/config/rc.local`? systemd .socket unit? Can you document it somewhere (README here? or in vmm-xen-unified repo?) ?

fepitre committed:
Add qubes.PESign

Related to https://github.com/QubesOS/qubes-issues/issues/8206

fepitre commented on rpc/qubes.PESign:
```diff
@@ -0,0 +1,33 @@
+if [ "$actual_size" -eq $((100 * 1024 * 1024)) ]; then
+    echo "Input size is at least 100MB. Aborting."
+    exit 1
+fi
```
Thank you, I've updated it.

fepitre commented:
> @fepitre what is the recommended method for redirecting `/var/run/qubes-pesign` to qrexec? socat started in `/rw/config/rc.local`? systemd .socket unit? Can you document it somewhere (README here? or in vmm-xen-unified repo?) ?

Doc is in vmm-xen-unified repo

marmarek committed:
PESign: improve logging and rename the service

Disable 'set -x' - it may leak some unintentional data in a future
version.
Print error message to stderr instead of stdout - otherwise it would be
taken as a signed payload.

And also rename the service to qubesbuilder.PESign.

QubesOS/qubes-issues#8206

marmarek committed:
PESign: support per-certificate parameters

Some certificates (especially stored on a hardware token) may need extra
parameters for `pesign`. Support those by sourcing
`~/.config/qubes-pesign/$CERTIFICATE`. The filename is safe as it's
sanitized by qrexec already. This also allows more flexible certificat
name handling - the config can override CERTIFICATE variable completely,
so the qrexec argument may be a friendly nickname independent of the
actual certificate name in the database (which is also a nickname, not
necessarily full DN).

For tokens, the config may look like this:

    # dbpath with pkcs11 module configured
    PESIGN_ARGS+=( "--certdir=$HOME/pesign-token-db" )
    # token name
    PESIGN_ARGS+=( "--token=token name" )
    # pinfile path, if relevant
    PESIGN_ARGS+=( "--pinfile=$HOME/pesign-token-pin.txt" )
    # you can also override CERTIFICATE
    CERTIFICATE="certificate name as on the token"

QubesOS/qubes-issues#8206

marmarek committed:
Document setup needed for UEFI Secureboot signing

This includes how to setup builder disposable template, socket->qrexec
proxy and all relevant parameters
This moves some of the setup steps from README.md in vmm-xen-unified
repo.

QubesOS/qubes-issues#8206

marmarek committed:
Add vmm-xen-unified to example configs

But don't build it by default, as it requires extra setup (signing keys
specifically).

QubesOS/qubes-issues#8206

DemiMarie commented on rpc/qubesbuilder.PESign:
```diff
@@ -0,0 +1,43 @@
+head --bytes=100MB > "$payload"
+
+# We don't allow payload being at least 100MiB
+if [ "$(stat --format=%s "$payload")" -ge $((100 * 1024 * 1024)) ]; then
```
```suggestion
size=$(stat --format=%s "$payload")
if [ "$size" -ge "$((100 * 1024 * 1024))" ]; then
```

Also please include a call to the PE checker I wrote.

marmarek replied:
link?

DemiMarie replied:
https://github.com/QubesOS/qubes-pecheck

fepitre commented on README.md:
```diff
@@ -131,6 +131,88 @@ $ qvm-prefs work-qubesos default_dispvm qubes-builder-dvm
+mkdir -p /rw/bind-dirs/etc/systemd/system/
+mkdir -p /usr/local/etc/default
+# adjust value if you used different key nickname, replace spaces with __
+echo 'KEY_NAME="Qubes__OS__Unified__Kernel__Image__Signing__Key"' > /usr/local/etc/default
```
```suggestion
echo 'KEY_NAME="Qubes__OS__Unified__Kernel__Image__Signing__Key"' > /usr/local/etc/default/qubes-pesign
```
